### ISSUE TITLE
Accessibility fixes for docs site: Mobile menu keyboard trap

### ISSUE THREAD

**Issue by n-dorman** [newscorp-ghfb/NewsKit#736]
### Description

The close button on the mobile menu can’t be accessed by keyboard. When you expand the menu you cannot get back to the close button using just the keyboard. We’re relying on the user hitting escape. 

https://user-images.githubusercontent.com/53819996/226962956-0ce0531f-bac4-4a11-b60e-1b028b952320.mp4

### Task

Include the close button in the menu focus order, after the user tabs through all the menu options.

### A/Cs

The user can access the menu close button using keyboard alone

Review with Design
